src/org/traccar/MainEventHandler.java

```diff
@@ -56,24 +60,43 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
             }
 
             String uniqueId = Context.getIdentityManager().getById(position.getDeviceId()).getUniqueId();
-
             // Log position
             StringBuilder s = new StringBuilder();
             s.append(formatChannel(ctx.channel())).append(" ");
             s.append("id: ").append(uniqueId);
-            s.append(", time: ").append(DateUtil.formatDate(position.getFixTime(), false));
-            s.append(", lat: ").append(String.format("%.5f", position.getLatitude()));
-            s.append(", lon: ").append(String.format("%.5f", position.getLongitude()));
-            if (position.getSpeed() > 0) {
+            if (logEvents.contains("time")) {
+                s.append(", time: ").append(DateUtil.formatDate(position.getFixTime(), false));
+            }
+            if (logEvents.contains("position")) {
+                s.append(", lat: ").append(String.format("%.5f", position.getLatitude()));
+                s.append(", lon: ").append(String.format("%.5f", position.getLongitude()));
+            }
+            if (position.getSpeed() > 0 && logEvents.contains("speed")) {
                 s.append(", speed: ").append(String.format("%.1f", position.getSpeed()));
             }
-            s.append(", course: ").append(String.format("%.1f", position.getCourse()));
-            if (position.getAccuracy() > 0) {
+            if (logEvents.contains("course")) {
+                s.append(", course: ").append(String.format("%.1f", position.getCourse()));
+            }
+            if (position.getAccuracy() > 0 && logEvents.contains("accuracy")) {
                 s.append(", accuracy: ").append(String.format("%.1f", position.getAccuracy()));
             }
-            Object cmdResult = position.getAttributes().get(Position.KEY_RESULT);
-            if (cmdResult != null) {
-                s.append(", result: ").append(cmdResult);
+            if (position.getOutdated() && logEvents.contains("outdated")) {
+                s.append(", outdated");
+            }
+            if (!position.getValid() && logEvents.contains("invalid")) {
+                s.append(", invalid");
+            }
+            if (logEvents.contains(Position.KEY_BATTERY_LEVEL)) {
```

Comment:
Maybe we should make it even more generic and support all attributes? It seems weird to support only some.

Reply:
maybe. I was thinking about doing something even more general, to be able to set a format string similar to log4j or apache, something like:
<entry key='logger.format'>pos: \{{position.lat},{position.lng}\} battery: {batteryLevel}%, {steps} steps</entry>
maybe I'll add it in a later pull request